### PR TITLE
Adding handling for Wolfram API errors

### PR DESCRIPTION
**Pull request by krachwal**
Added handling for successful Wolfram API errors (the response has embedded error flagging but returns successful when it fails to come up with a result for any reason). 
Added a custom error and API tests (which ran as part of the actions/checks on this PR). 

Also added some docstrings and separated coordinate functions from math functions.

### CODE REVIEW

File: python-backend/wolfram_client.py [hunk 1]
```diff
@@ -36,8 +41,17 @@ def ask(query: str, include_pod: str = None) -> str:
...
-            except Exception as e:
+                result = json.loads(result.content)
+                try:
+                    result["queryresult"]["error"]["code"]
+                except (AttributeError, TypeError):
+                    # if the key does not exist or is false then the api call was successful
+                    pass
+                else:
+                    raise APIError(result["queryresult"]["error"]["msg"])
+                print("TYPE = {}", type(result))
```
Comment (davidbrownell) [inline python-backend/wolfram_client.py:52]: Should this be using a logger rather than print?

File: python-backend/wolfram_client.py [hunk 2]
```diff
@@ -36,8 +41,17 @@ def ask(query: str, include_pod: str = None) -> str:
...
-                return json.loads(result.content)
-            except Exception as e:
+                result = json.loads(result.content)
+                try:
+                    result["queryresult"]["error"]["code"]
+                except (AttributeError, TypeError):
+                    # if the key does not exist or is false then the api call was successful
+                    pass
+                else:
+                    raise APIError(result["queryresult"]["error"]["msg"])
```
Comment (davidbrownell) [inline python-backend/wolfram_client.py:51]: I had to lookup this syntax :)

Can this statement be moved to follow the statement on line 46 to achieve the same behavior?

File: python-backend/wolfram_client.py (hunk 1 above)
Comment (krachwal) [reply python-backend/wolfram_client.py:52]: that should actually be gone - it was form my testing.

File: python-backend/math_utils.py
Comment (ketanbj) [inline python-backend/math_utils.py:1]: I am glad that you removed delta_coordinates(). May be personal preference but one line return with nested conditions and loops make my head hurt.

**Review by ketanbj — COMMENTED** [review]
Look good to me and passing all the checks. I say ship it.

**Review by ketanbj — APPROVED** [review]

File: python-backend/wolfram_client.py (hunk 2 above)
Comment (krachwal) [reply python-backend/wolfram_client.py:51]: well, apparently it can!

**Review by davidbrownell — APPROVED** [review]

**Review by ketanbj — APPROVED** [review]

**Pull request merged**